trieste/acquisition/multi_objective/partition.py

```diff
@@ -64,46 +89,83 @@ def __post_init__(self) -> None:
         tf.debugging.assert_shapes([(self.lower_idx, ["N", "D"]), (self.upper_idx, ["N", "D"])])
 
 
-class Pareto:
+class BoundIndexPartition(NonDominatedPartition):
     """
-    A :class:`Pareto` Construct a Pareto set.
-    Stores a Pareto set and calculates the cell bounds covering the non-dominated region.
-    The latter is needed for certain multiobjective acquisition functions.
-
-    For hypervolume-based multiobjective optimisation with n>2 objectives, this class
-    implements branch and bound procedure algorithm. a divide and conquer method introduced
-    in :cite:`Couckuyt2012`.
+    A collection of partition strategy that is based on storing the index of pareto fronts
+    & other auxiliary points
     """
 
-    def __init__(self, observations: TensorType, *, jitter: float = DEFAULTS.JITTER):
+    front: TensorType
+    _bounds: _BoundedVolumes
+
+    def partition_bounds(
+        self, anti_reference: TensorType, reference: TensorType
+    ) -> tuple[TensorType, TensorType]:
         """
-        :param observations: The observations for all objectives, with shape [N, D].
-        :raise ValueError (or InvalidArgumentError): If ``observations`` has an invalid shape.
+        Get the partitioned hypercell's lower and upper bounds.
+
+        :param anti_reference: a worst point to use with shape [D].
+            Defines the lower bound of the hypercell
+        :param reference: a reference point to use, with shape [D].
+            Defines the upper bound of the hypervolume.
+            Should be equal to or bigger than the anti-ideal point of the Pareto set.
+            For comparing results across runs, the same reference point must be used.
+        :return: lower, upper bounds of the partitioned cell
+        :raise ValueError (or `tf.errors.InvalidArgumentError`): If ``reference`` has an invalid
+            shape.
         """
-        tf.debugging.assert_rank(observations, 2)
-        tf.debugging.assert_greater_equal(tf.shape(observations)[-1], 2)
+        tf.debugging.assert_greater_equal(reference, self.front)
+        tf.debugging.assert_greater_equal(self.front, anti_reference)
+        tf.debugging.assert_type(anti_reference, self.front.dtype)
+        tf.debugging.assert_type(reference, self.front.dtype)
```

Comment:
Are we testing all these in the unit test?

Reply:
yes for 117 and 118 but not for dtype related (119-120), not sure do we need this dtype check in test?

Reply:
@uri-granta  any option?

Reply:
I'm not too concerned, though it's easy enough to add the tests.

Reply:
i'm a bit cautious about to put test for type here, one hand it is just written defensively and easy to interpret (perhaps too easy for a test and not necessary?), the other hand is if we'd keep type test like this in `Partition`, each partition method (that may come up in future) will have to align with this type test as well (means big chuck of codes in test just for type). Please let me know if there are any opinion on type tests here.